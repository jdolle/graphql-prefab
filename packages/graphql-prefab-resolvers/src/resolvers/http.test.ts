import moxios from 'moxios'
import httpResolver from './http'

it('#httpResolver throws if url is undefined', () => {
  expect(httpResolver()()).rejects.toMatchSnapshot()
  expect(httpResolver({})()).rejects.toMatchSnapshot()
})

it('#httpResolver calls axios', (done) => {
  moxios.withMock(() => {
    const result = httpResolver({ url: '/example' })()

    moxios.wait(async () => {
      const request = moxios.requests.mostRecent()
      await request.respondWith({ response: ['foo'] })
      expect(request.url).toBe('/example')
      expect(result).resolves.toEqual(['foo'])
      done()
    })
  })
})
