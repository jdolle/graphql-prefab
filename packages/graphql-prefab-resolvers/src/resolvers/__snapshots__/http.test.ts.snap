// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#httpResolver throws if url is undefined 1`] = `[Error: url required]`;

exports[`#httpResolver throws if url is undefined 2`] = `[Error: url required]`;
